pkg/providers/imagefamily/agentbakerbootstrap/provisionclientbootstrap.go

```diff
@@ -0,0 +1,239 @@
+			SSHAccess: lo.ToPtr(models.SSHAccess_LocalUser),
+			// EnableVTPM:       lo.ToPtr(false), // Unsupported as of now (Trusted launch)
+			// EnableSecureBoot: lo.ToPtr(false), // Unsupported as of now (Trusted launch)
+		},
+
+		// VnetCidrs:               []string{},                                      // Unsupported as of now; TODO(Windows)
+		// MessageOfTheDay:         lo.ToPtr(""),                                    // Unsupported as of now
+		// AgentPoolWindowsProfile: &models.AgentPoolWindowsProfile{},               // Unsupported as of now; TODO(Windows)
+		// KubeletDiskType:         lo.ToPtr(models.KubeletDiskType_Unspecified),    // Unsupported as of now
+		// CustomLinuxOSConfig:     &models.CustomLinuxOSConfig{},                   // Unsupported as of now (sysctl)
+		// EnableFIPS:              lo.ToPtr(false),                                 // Unsupported as of now
+		// GpuInstanceProfile:      lo.ToPtr(models.GPUInstanceProfile_Unspecified), // Unsupported as of now (MIG)
+		// WorkloadRuntime:         lo.ToPtr(models.WorkloadRuntime_Unspecified),    // Unsupported as of now (Kata)
+		// ArtifactStreamingProfile: &models.ArtifactStreamingProfile{
+		// Enabled: lo.ToPtr(false), // Unsupported as of now
+		// },
+	}
+
+	if p.Arch == "amd64" {
+		provisionProfile.Architecture = lo.ToPtr("x64")
+	} else {
+		provisionProfile.Architecture = lo.ToPtr(p.Arch)
+	}
+
+	switch p.ImageFamily {
+	case v1alpha2.Ubuntu2204ImageFamily:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_Ubuntu)
+	case v1alpha2.AzureLinuxImageFamily:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_AzureLinux)
+	default:
+		provisionProfile.OsSku = to.Ptr(models.OSSKU_Ubuntu)
+	}
+
+	if p.KubeletConfig != nil {
+		provisionProfile.CustomKubeletConfig = &models.CustomKubeletConfig{
+			// AllowedUnsafeSysctls: ..., // Unsupported as of now
+			CPUCfsQuota: p.KubeletConfig.CPUCFSQuota,
+		}
+	}
+
+	if p.KubeletConfig != nil && p.KubeletConfig.MaxPods != nil {
+		provisionProfile.MaxPods = p.KubeletConfig.MaxPods
+	} else {
+		provisionProfile.MaxPods = lo.ToPtr(int32(250)) // Delegatable defaulting?
+	}
+
+	if modeString, ok := p.Labels["kubernetes.azure.com/mode"]; ok && modeString == "system" {
+		provisionProfile.Mode = lo.ToPtr(models.AgentPoolMode_System)
+	} else {
+		provisionProfile.Mode = lo.ToPtr(models.AgentPoolMode_User)
+	}
+
+	if utils.IsNvidiaEnabledSKU(p.InstanceType.Name) {
+		if utils.UseGridDrivers(p.InstanceType.Name) {
+			provisionProfile.GpuProfile = &models.GPUProfile{
+				DriverType:       lo.ToPtr(models.DriverType_GRID),
+				InstallGPUDriver: lo.ToPtr(true),
+			}
+		} else {
+			provisionProfile.GpuProfile = &models.GPUProfile{
+				DriverType:       lo.ToPtr(models.DriverType_CUDA),
+				InstallGPUDriver: lo.ToPtr(true),
+			}
+		}
+
+	}
+
+	if p.IsWindows {
+		provisionProfile.OsType = lo.ToPtr(models.OSType_Windows)
+	} else {
+		provisionProfile.OsType = lo.ToPtr(models.OSType_Linux)
+	}
```

Comment:
Seems odd we need this at all since we pass in OS SKU. This alongside some of these other values could be easily infered in a more minimal AgentpoolProfile payload